_posts/2018-04-01-kudos-mapping-and-slam.md

```diff
@@ -0,0 +1,38 @@
+---
+layout: post
+title: Kudos Mapping and SLAM
+date: 2018-04-01
+categories: 
+    - Robotics
+    - SLAM
+    - ROS
+    - Mapping
+description: Added LiDAR mapping functionality to Computer Science House's floor robot, Kudos.  
+image: https://github.com/ComputerScienceHouse/Kudos/blob/master/kudos_csh/images/CSH_Map_2.png?raw=true
+image-sm: http://www.trevorsherrard.com/images/CSH_Map_2.png
+author: Trevor Sherrard
+author-image: http://www.trevorsherrard.com/images/profile_pics.jpg
+author-bio: Electrical Engineer, roboticist, computer vision enthusiast. I have a strong passion seeing my software come alive in my projects.
+author-email: tws4129@rit.edu
+author-social:
+    github: https://github.com/sherrardTr4129
+    linkedin: https://www.linkedin.com/in/trevor-sherrard
+---
+
+## Background
+Computer Science House purchased a robot chasis from one of our members in the Spring semester of 2017. I had worked on previous versions of the CSH H.E.R.O.I.N project and thought I would be able to make significant strides given we now had a stable robot chasis. So, using robot operating system constructs and several new pieces of hardware/sensors, I added mapping functionality to Kudos.
+
+## Details
+First, I needed to install a LiDAR sensor on the robot. Standard off the self LiDAR solutions, such as the 16-channel velodyne puck, start at upwards of $5000. Being a student organization, we needed a more financially sensible solution. I decided on using a replacement LiDAR sensor from the NeatoXV robot vaccum cleaner. Once the serial communication between the LiDAR controller board was setup with the Raspberry Pi, ROS development began. The LiDAR scans were "off-boarded" the robot via the rostopic /laserscan via the network to a VM running on CSH proxmox. On this VM, a program subscribed to this data and performed an algorithm to simultaneously localize the robot within a map and build the map as the robot moved through its enviornment. A navigation goal could be set for the robot to navigate to, and a path to follow to obtain this goal was computed. From this path, individual robot movements were calculated and streamed to the robot via Computer Science House's WiFi network. This robot movement was actuated upon by the robot through its move base controller. See the figure below for a drawing of the control topology for Kudos.
```

Comment:
`s/off the self/off-the-shelf/`
`s/velodyne/Velodyne/`
`s/vaccum/vacuum/`
`s/"off-boarded" the robot/"off-boarded" from the robot/`
Using "via" twice in a row like this is confusing, I'd try to reword this.
`s/proxmox/Proxmox/`